jumanji/environments/combinatorial/cvrp/env_test.py

```diff
@@ -0,0 +1,305 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import chex
+import jax
+import pytest
+from jax import numpy as jnp
+from jax import random
+
+from jumanji.environments.combinatorial.cvrp.env import CVRP
+from jumanji.environments.combinatorial.cvrp.types import State
+from jumanji.environments.combinatorial.cvrp.utils import (
+    DEPOT_IDX,
+    compute_tour_length,
+    get_augmentations,
+)
+
+from jumanji.testing.env_not_smoke import check_env_does_not_smoke
+from jumanji.testing.pytrees import assert_is_jax_array_tree
+from jumanji.types import StepType, TimeStep
+
+
+@pytest.fixture
+def cvrp_env() -> CVRP:
+    """Instantiates a default CVRP environment."""
+    return CVRP()
+
+
+def test_cvrp__reset(cvrp_env: CVRP) -> None:
+    """Validates the jitted reset of the environment."""
+    reset_fn = jax.jit(cvrp_env.reset)
+    key = random.PRNGKey(0)
+    state, timestep = reset_fn(key)
```

Comment:
Could we add a check for compilation here? Check that calling reset twice does not recompile (e.g. if you set one of the input as static). Please see the [BinPack tests](https://github.com/instadeepai/jumanji/blob/main/jumanji/environments/combinatorial/binpack/env_test.py#L136) for instance, for an example of how to do.
```suggestion
    chex.clear_trace_counter()
    reset_fn = jax.jit(chex.assert_max_traces(cvrp_env.reset, n=1))

    key = jax.random.PRNGKey(0)
    state, timestep = reset_fn(key)
    # Call again to check it does not compile twice.
    state, timestep = reset_fn(key)
```